astropy/modeling/parameters.py

```diff
@@ -308,6 +387,47 @@ def value(self, value):
         self._set_model_value(self._model, value)
 
     @property
+    def unit(self):
+        """
+        The unit attached to this parameter, if any.
+
+        On unbound parameters (i.e. parameters accessed through the
+        model class, rather than a model instance) this is the required/
+        default unit for the parameter.
+        """
+
+        if self._model is None:
+            return self._unit
+        else:
+            # orig_unit may be undefined early on in model instantiation
+            return self._model._param_metrics[self.name].get('orig_unit',
+                                                             self._unit)
+
+    @unit.setter
+    def unit(self, unit):
+        if self._model is None:
+            raise AttributeError('Cannot set unit on a parameter definition')
+
+        # TODO: This isn't really the right thing to do, and is just a
```

Comment:
If this is effectively for display/convenience to the user, shouldn't one also convert the value to the new unit? If one also converts the default, it may solve the problem with keeping track of the unit of the default as well.